content/5_Plat/HLD.mdx

```diff
@@ -15,6 +15,253 @@ frequency: 1
+  at the child.
+- A **light child** of a node is any child that is not a **heavy child**.
+- A **heavy edge** connects the current node to its **heavy child**.
+- A **light edge** connects the current node to any of its **light children**.
+- A **heavy path** is the path formed by a collection **heavy edges**.
+- A **light path** is the path formed by a collection **light edges**.
+
+## Properties:
+
+Given the definitions stated above, an important property can be derived (I will
+prove this a bit later).
+
+- Any path from node $x$ to node $y$ on the tree can pass through at most
+  $\mathcal{O}(\log N)$ light edges.
+
+Lets suppose the above property is true for a moment. Since a heavy path can
+only be broken by a light edge (or else the edge will be a part of the heavy
+path), we can know that there are at most $\mathcal{O}(\log N)$ heavy paths on
+any path from an arbitrary node $x$ to an arbitrary node $y$.
+
+In addition, by using segment trees (or any other RURQ data structure) we can
+calculate the value of any consecutive interval on any heavy path in
+$\mathcal{O}(\log N)$ time.
+
+Since there are at most $\mathcal{O}(\log N)$ heavy paths and
+$\mathcal{O}(\log N)$ light edges, computing the value on the path from node $x$
+to node $y$ will take $\mathcal{O}(\log^2 N + \log N)$ = $\mathcal{O}(\log^2 N)$
+time. We can answer $Q$ queries in $\mathcal{O}(Q \log^2 N)$ time if the above
+property is true.
+
+## Proof:
+
+Finally, we would need to prove the property above to conclude the tutorial.
+Consider a path from node $x$ to node $y$, this path can be broken down into a
+path from node $x$ to $LCA(x,y)$ and another path from node $y$ to $LCA(x,y)$.
+$LCA(x,y)$ must be an ancestor of both $x$ and $y$ by definition. Hence now we
+only need to prove that the path from any node to any of its ancestors can only
+pass through $\mathcal{O}(\log N)$ light edges to prove the property above.
+
+Suppose that a light edge connects node $x$ and node $y$, where $y$ is a child
+of $x$. By the definition of light edge, $y$ must be a light child of $x$. This
+means that there has to be another child of $x$ with a larger subtree size than
+$y$, or else $y$ will become the heavy child of $x$. Hence the size of the
+subtree rooted at $x$ must be at least two times of the size of the subtree
+rooted at $y$. Since the subtree size of a node can not exceed the total number
+of nodes $N$, the doubling process can only happen at most $\mathcal{O}(\log N)$
+times when traversing up from an arbitrary node to its parent along a light
+edge. Given this, we now know that there could be at most $\mathcal{O}(\log N)$
+light edges in both the path from $x$ to $LCA(x,y)$ and the path from $LCA(x,y)$
+to $y$. Hence there are at most $\mathcal{O}(\log N + \log N)$ =
+$\mathcal{O}(\log N)$ light edges in the path from an arbitrary node $x$ to an
+arbitrary node $y$.
+
+We can solve the focus problem "Subtrees & Paths" using Heavy Light
```

Comment:
Subtree & Paths is the same as the first CSES problem in the practice problems. There's already a code solution for that at https://usaco.guide/problems/cses-2134-path-queries-ii/solution (which I prefer), so could you do the following please:
- Replace the HR focus problem with the CSES problem
- Copy over the existing code
- Make it look nicer by adding appropriate spacing and/or comments